qubesfiletrust/qvm_file_trust.py

```diff
@@ -0,0 +1,546 @@
+                safe_chmod(path, orig_perms,
+                    'Unable to set original perms. on {}'.format(path))
+
+        # Remove visual attributes
+        set_visual_attributes(path, False)
+
+    else:
+        # Set file to untrusted
+        # AKA add our xattr and lock
+        try:
+            xattr.setxattr(path, 'user.qubes.untrusted', 'true')
+            safe_chmod(path, 0o0,
+                    'Unable to set untrusted permissions on: {}'.format(path))
+        except:
+            # Unable to remove our xattr, return original permissions
+            safe_chmod(path, orig_perms,
+                'Unable to return perms after setting as untrusted: {}'.
+                format(path))
+            sys.exit(65)
+
+        # Add visual attributes
+        set_visual_attributes(path, True)
+
+def change_folder(path, trusted):
+    """Change the trust state of a folder"""
+
+    # Remove '/' from end of path
+    path = os.path.normpath(path)
+
+    try:
+        # Create the ~/.config/qubes folder if it doesn't exist
+        home_dir = os.path.expanduser('~')
+        if not os.path.exists(home_dir + '/.config'):
+            os.mkdir(home_dir + '/.config')
+        if not os.path.exists(home_dir + '/.config/qubes'):
+            os.mkdir(home_dir + '/.config/qubes')
+
+        # Create the local file if it does not exist
+        if not os.path.exists(LOCAL_FOLDER_LOC):
+            open(LOCAL_FOLDER_LOC, 'a').close()
+    except:
+        error('Could not create local rule list: {}'.format(
+                LOCAL_FOLDER_LOC))
+        error('Check /home/<your user> folder exists...')
+        sys.exit(72)
+
+    if trusted:
+        # Set folder to trusted
+        # AKA remove any mentions from untrusted paths list
+        try:
+            file = open(LOCAL_FOLDER_LOC, 'r+')
+        except:
+            error('Unable to read local untrusted folder: {}'.
+                    format(LOCAL_FOLDER_LOC))
+            sys.exit(72)
+
+        lines = file.readlines()
+        file.seek(0)
+
+        # Write back all lines to the file except ones containing our path
+        for line in lines:
+            if line.rstrip() != path:
+                file.write(line)
+
+        file.truncate()
+        file.close()
+
+        # Remove visual attributes
+        set_visual_attributes(path, False)
+    else:
+        # Set folder to untrusted
+        # AKA add path to untrusted paths list
+
+        # Ensure path isn't already in untrusted paths list
+        try:
+            with open(LOCAL_FOLDER_LOC) as local_list:
+                for line in local_list:
+                    if line.rstrip() == path:
+                        # Already untrusted, just return
+                        return
+        except:
+            error('Unable to read local untrusted folder: {}'.
+                    format(LOCAL_FOLDER_LOC))
+            sys.exit(72)
+
+        # Append path to the bottom
+        file = open(LOCAL_FOLDER_LOC, 'ab')
```

Comment:
1. use `with`
2. since you have `unicode` data to write anyway, `b` shouldn't be needed

Reply:
Corrected in 45441f2